### ISSUE TITLE
va-accordion: Headline slots prevent keypress activation in NVDA

### ISSUE THREAD

**Issue by amyleadem** [department-of-veterans-affairs/vets-design-system-documentation#4851]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

In NVDA, the accordion buttons in the [Headline slot variant](https://design.va.gov/storybook/?path=/story/uswds-va-accordion--headline-slot) cannot be activated with standard keypresses like `enter` or`spacebar`. 

This issue appears to be isolated keypresses in NVDA when using the headline slot. Specifically, this issue is NOT happening:
- In accordions that set the headline with the header prop
- In NVDA when the button is activated via mouse click
- in JAWS or VoiceOver

### Details
Currently, using the headline slot creates a nested header structure inside `usa-accordion__heading`. (See the code samples in this section for examples.)

This causes a a few problems:
1. It creates duplicate headers in the DOM, which can be confusing and cumbersome for screen reader users
2. It causes a conflict between the `level` prop and the headline slot heading level (More details in the [related issue about the level prop conflict](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4636))
3. It causes issues with core functionality in NVDA, where activating the accordion item with `enter` or `spacebar` does not work

#### Current rendered markup for accordion heading element:
_Note how the H6 is inside H3_
```html
<h3 class="usa-accordion__heading">
    <button type="button" class="usa-accordion__button" aria-expanded="true" aria-controls="content" part="accordion-header">
        <span class="va-accordion__header">
            <slot name="headline"></slot> // ↪️ <h6 slot="headline">Slotted First Amendment</h6>
        </span>
    </button>
</h3>
```

#### Ideal rendered markup
_Note that the `usa-accordion__heading` element uses the slot's h6 and there are no headers inside `usa-accordion__heading`_ 
```html
<h6 class="usa-accordion__heading">
    <button type="button" class="usa-accordion__button" aria-expanded="true" aria-controls="content" part="accordion-header">
        <span class="va-accordion__header">
            Slotted First Amendment
        </span>
    </button>
</h6>
```

## Reproducing

- CSS Library version:
- Device: (e.g. iPhone 8, Macbook)
- Browser: (e.g. Firefox, IE 11)

**Steps to reproduce:**

1. Open the headline [slot accordion story](https://design.va.gov/storybook/?path=/story/uswds-va-accordion--headline-slot)
2. Turn on NVDA
3. Tab to an accordion item button
4. Try to expand the accordion item by pressing the `enter` or `space` key. 
5. Notice that the component does not respond to the keypress

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

**Comment by amyleadem** [comment]
@caw310 
To investigate this issue, I needed to draft some solution code to make sure I could reliably reproduce the issue. I ended up creating a [draft PR](https://github.com/department-of-veterans-affairs/component-library/pull/1788) from this demo/test code, and would be interested in scheduling this once my other sprint tasks are complete.
